ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -817,26 +826,21 @@ func (r *RayServiceReconciler) getAndCheckServeStatus(ctx context.Context, dashb
 				HealthLastUpdateTime: &timeNow,
 			}
 
-			if deployment.Status != rayv1alpha1.DeploymentStatusEnum.HEALTHY {
+			if deployment.Status == rayv1alpha1.DeploymentStatusEnum.UNHEALTHY {
 				prevStatus, exist := prevApplicationStatus.Deployments[deploymentName]
 				if exist {
-					if prevStatus.Status != rayv1alpha1.DeploymentStatusEnum.HEALTHY {
+					if prevStatus.Status == rayv1alpha1.DeploymentStatusEnum.UNHEALTHY {
 						deploymentStatus.HealthLastUpdateTime = prevStatus.HealthLastUpdateTime
-
-						if !isHealthy || (prevStatus.HealthLastUpdateTime != nil && time.Since(prevStatus.HealthLastUpdateTime.Time).Seconds() > serviceUnhealthySecondThreshold) {
-							// TODO (kevin85421): Without `!isHealthy`, this `if` statement is almost impossible to be reached because the `HealthLastUpdateTime` of a serve deployment
-							// is always later than the `HealthLastUpdateTime` of the serve application. Hence, the restart is always triggered by the serve application. If we
-							// can confirm that `isHealthy = false` is always set by the serve application check, we can remove the `time.Since` check here.
+						if !isHealthy {
 							r.Log.Info("Restart RayCluster", "deploymentName", deploymentName, "appName", appName, "restart reason",
 								fmt.Sprintf(
-									"The status of the serve deployment %s or the serve application %s has not been HEALTHY/RUNNING for more than %f seconds. "+
+									"The serve application %s has been UNHEALTHY or DEPLOY_FAILED for more than %f seconds. "+
+										"This may be caused by the serve deployment %s is UNHEALTHY. "+
```

Comment:
```suggestion
										"This may be caused by the serve deployment %s being UNHEALTHY. "+
```
Will the user be able to tell from the logs which of the following scenarios happened?
- application X was unhealthy for too long
- application X was deploy_failed for too long
- deployment Y was unhealthy for too long

I understand there's some overlap here so it might be hard to capture in a single message at the end.  But (for example) if we logged these statuses on each health check iteration, the user would probably be able to figure it out.

Reply:
> But (for example) if we logged these statuses on each health check iteration

We have already logged the information for each health check iteration.

https://github.com/ray-project/kuberay/blob/cc6b7ba832a8d80283a43cfbbf61950d7caab0b0/ray-operator/controllers/ray/rayservice_controller.go#L770

Reply:
https://github.com/ray-project/kuberay/pull/1293/commits/413a8038da2a090248a1af1d9c0c59ec4b4078e9

=> Print the status of the Serve application when it is considered as unhealthy.